parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java

```diff
@@ -228,7 +232,7 @@ private static void generateUnpack(FileWriter fw, int bitWidth, int batch, boole
           } else if (shift > 0){
             shiftString = "<<  " + shift;
           }
-          fw.append(" (((((int)in[" + align(byteIndex, 2) + " + inPos]) & 255) " + shiftString + ") & " + mask + ")");
+          fw.append(" (((((int)in.get(" + align(byteIndex, 2) + " + inPos)) & 255) " + shiftString + ") & " + mask + ")");
         }
         fw.append(";\n");
       }
```

Comment:
Fixed, added a boolean parameter to the method to specify if a byte array should be used. Where the method is being called I added a true/false combination for both uses of the method to generate 4 total methods.